### PR TITLE
develop: refactor MySQL Connector/Python dev guide

### PR DESCRIPTION
**Pull request by wd0517**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-python-mysql-connector.md [hunk 1]
```diff
@@ -1,293 +1,282 @@
...
-        try:
-            trade_update(cursor, sell_id, buy_id, amount, price)
-        except Exception as err:
-            connection.rollback()
-            print(f'something went wrong: {err}')
-        else:
-            connection.commit()
-            print("trade success")
+    ```dotenv
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-mysql-connector.md:134]:
```suggestion
    TIDB_HOST='{host}'
```

File: develop/dev-guide-sample-application-python-mysql-connector.md [hunk 2]
```diff
@@ -1,293 +1,282 @@
...
+    TIDB_PORT='4000'
+    TIDB_USER='{username}'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-mysql-connector.md:136]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-python-mysql-connector.md
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysql-connector.md:1]: Please update the link text in `/develop/dev-guide-choose-driver-or-orm.md` and `/develop/dev-guide-insert-data.md`:

```
For an example of using MySQL Connector/Python to build a TiDB application, see [Connect to TiDB with MySQL Connector/Python](/develop/dev-guide-sample-application-python-mysql-connector.md#step-2-get-the-code).
```

**Review by Oreoxmt — APPROVED** [review]

**Review by sykp241095 — APPROVED** [review]

**Pull request merged**